packages/recommend-react/src/useFrequentlyBoughtTogether.ts

```diff
@@ -29,35 +31,84 @@ export function useFrequentlyBoughtTogether<TObject>({
   const objectIDs = useStableValue(userObjectIDs);
   const queryParameters = useStableValue(userQueryParameters);
 
-  useAlgoliaAgent({ recommendClient });
+  const { hasProvider, register } = useRecommendContext();
+  const { client, isContextClient } = useRecommendClient(recommendClient);
+
+  useAlgoliaAgent({ recommendClient: client });
 
   const transformItemsRef = useRef(userTransformItems);
   useEffect(() => {
     transformItemsRef.current = userTransformItems;
   }, [userTransformItems]);
 
   useEffect(() => {
-    setStatus('loading');
-    getFrequentlyBoughtTogether({
+    const param = {
       indexName,
       maxRecommendations,
       objectIDs,
       queryParameters,
-      recommendClient,
       threshold,
       transformItems: transformItemsRef.current,
-    }).then((response) => {
-      setResult(response);
-      setStatus('idle');
-    });
+    };
+    let unregister: Function | undefined;
```

Comment:
is unregister used somewhere else than in the return of the effect? if not it can be moved to the usage and made const

Reply:
`unregister` is defined conditionally, how would you move it in this case? 

```js
  useEffect(() => {
    let unregister;
    if(isBatch){
      unregister = register({...})
    }else{
      // do something else
    }
    return ()=>{
      if(unregister){
        unregister()
      }
    }
  }, [])
```

Reply:
ah the file is too long, I did not catch that the return is in a different level.

However what you can do instead is:

```
  useEffect(() => {
    if(isBatch){
      return register({...})
    }

    // do something else
    return () => {}
  }, [])
```

which simplifies and lowers the levels of conditionality needed.

Reply:
updated 👍